src/app/layout/navbar/language-button/language-button.service.ts

```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@angular/core';
+
+import { TranslateService } from '@ngx-translate/core';
+import { Observable, Subject } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LanguageButtonService {
+  changeLanguage$: Observable<any>;
+  currentLanguage: string;
+  private languageChanged: Subject<any>;
+
+  constructor(private translateService: TranslateService) {
+    this.languageChanged = new Subject<any>();
+    this.changeLanguage$ = this.languageChanged.asObservable();
+  }
+
+  setDefaultLanguage(): void {
+    this.translateService.setDefaultLang('en');
+    this.switchLanguage(this.translateService.getBrowserLang());
+  }
+
+  switchLanguage(language: string): void {
+    this.currentLanguage = language;
+    this.translateService.use(language);
+    this.languageChanged.next(language);
+  }
+}
```

Comment:
Maybe we should have a language service instead of a language button service. This service could be used in other components different to the language button component. Maybe this could be in the core module.